drivers/isl29125/isl29125.c

```diff
@@ -54,7 +59,31 @@ int isl29125_init(isl29125_t *dev, i2c_t i2c, gpio_t gpio,
 
     /* TODO: implement configuration 2: infrared compensation configuration */
 
-    /* TODO: implement configuration 3: interrupt mode configuration */
+    /* configuration 3: interrupt mode configuration */
+    uint8_t conf3 = 0x00;
+    conf3 |= interrupt_status;
+    conf3 |= interrupt_persist;
+    conf3 |= interrupt_conven;
+
+    /* Lower and higher interrupt threshold registers. */
+    uint8_t lthlb = 0x00;
+    uint8_t lthhb = 0x00;
+    uint8_t hthlb = 0x00;
+    uint8_t hthhb = 0x00;
+    uint16_t max_range = 10000;
+
+    if (range == 0x00)
+        max_range = 375;
+
+    if ((lower_threshold <= max_range) && (higher_threshold <= max_range) && (lower_threshold < higher_threshold))
+    {
+        lower_threshold *= (uint16_t) (65535.0/max_range);
```

Comment:
Is floating precision really necessary? You are casting the result to `uint16_t` anyways, so I would vote to convert this to integer division by dropping the `.0`